Microsoft.Toolkit.Uwp.UI/Media/CanvasBrushBase.cs

```diff
@@ -38,12 +38,24 @@ public abstract class CanvasBrushBase : XamlCompositionBrushBase
         /// <returns>True if drawing was completed and the brush is ready, otherwise return False to not create brush yet.</returns>
         protected abstract bool OnDraw(CanvasDevice device, CanvasDrawingSession session, Vector2 size);
 
+        /// <summary>
+        /// Gets or sets the shared device.
+        /// </summary>
+        protected CanvasDevice Device { get; set; }
+
+        /// <summary>
+        /// Gets or sets the internal graphics composition.
+        /// </summary>
+        protected CompositionGraphicsDevice Graphics { get; set; }
+
         /// <summary>
         /// Initializes the Composition Brush.
         /// </summary>
         protected override void OnConnected()
         {
             base.OnConnected();
+            Device = CanvasDevice.GetSharedDevice();
```

Comment:
I feel like the registration of these shouldn't move, just unsubscribe/resubscribe to the events inline after you have them?

Device = CanvasDevice.GetSharedDevice();
Device.DeviceLost -= CanvasBrushBase_DeviceLost;
Device.DeviceLost += CanvasBrushBase_DeviceLost;